### PR TITLE
Make use of rust 2018 features

### PR DESCRIPTION
**Pull request by poros**
- Install rustfmt and clippy on stable
- Drop rustfmt option only available in nightly
- Remove all unecessary extern crate and macro use
- Update serde and failure crate usage of macros
- Optimize some imports
- Bump dependencies
- Also minor adjustments to Makefile

### CODE REVIEW

**Comment by poros** [comment]
They broke clippy in nightly 1.35. The recommended solution for now seems to be allowing travis to fail for nightly. Is it acceptable? Unfortunately, we can only run the benchmarks in nightly, so it means we will give up the benchmarks when nightly is broken (hopefully not often?)

File: README.md [hunk 1]
```diff
@@ -20,20 +20,17 @@ Add to your `Cargo.toml`:
 
 ```toml
 [dependencies]
-avro-rs = "^0.6"
+avro-rs = "0.6"
+failure = "0.1.5"
```
Comment (flavray) [inline README.md]: Why are these needed? (failure & serde) Won't they get pulled in alongside avro-rs when it's added as a dependency?

File: Makefile [hunk 2]
```diff
@@ -15,47 +14,45 @@ env: $(VENV)
...
+	cargo clippy --all-features
 
 # TESTING
 
 .PHONY: test
-test:
+test: install-hooks
 	cargo test --all-features
-
+	$(VENV)/bin/pre-commit run --all-files
```
Comment (flavray) [inline Makefile]: Not sure I approve of `make test` having side effects (namely changing code with `fmt`), but that's not a huge deal :)

File: README.md (hunk 1 above)
Comment (flavray) [reply README.md]: Heh, just saw #74 . Resolving this

File: Makefile (hunk 2 above)
Comment (poros) [reply Makefile]: Yeah, but how do we make sure people are formatting their code? Not everyone enables pre-commits.

**Pull request merged**